src/crypto/index.js

```diff
@@ -3028,19 +3029,28 @@ Crypto.prototype.encryptEvent = async function(event, room) {
  *  finished decrypting. Rejects with an `algorithms.DecryptionError` if there
  *  is a problem decrypting the event.
  */
-Crypto.prototype.decryptEvent = function(event) {
+Crypto.prototype.decryptEvent = async function(event) {
     if (event.isRedacted()) {
-        return Promise.resolve({
+        const redactionEvent = new MatrixEvent(event.getUnsigned().redacted_because);
+        const content = redactionEvent.getWireContent();
+        const alg = this._getRoomDecryptor(event.getRoomId(), content.algorithm);
+        const decryptedEvent = await alg.decryptEvent(redactionEvent);
```

Comment:
Instead of duplicating these lines, could you instead call this function recursively to decrypt the redaction event? Something like `this.decryptEvent(redactionEvent)`?